app/services/update_team_name_and_ldu_service.rb

```diff
@@ -0,0 +1,42 @@
+class UpdateTeamNameAndLduService
+  def self.update(team_code:, team_name:, ldu_code:)
+    new(team_code: team_code, team_name: team_name, ldu_code: ldu_code).update
+  end
+
+  def initialize(team_code:, team_name:, ldu_code:)
+    @team = Team.find_by(code: team_code)
+    @team_name = team_name
+    @ldu_code = ldu_code
+
+    Rails.logger.error("Couldn't find team with code #{team_code}") if @team.nil?
+  end
+
+  def update
+    return if @team.nil?
+
+    update_name if name_needs_updating?
+    update_ldu if ldu_needs_updating?
+    @team.save if @team.changed?
```

Comment:
Are you checking the name in this way because you're concerned about the extra time of updating records that haven't changed? Or because you wanted the same pattern for name / ldu?

The ActiveRecord dirty-tracker is pretty clever, you can safely update that attributes and `.changed?` will give you what you want, I think. I think it may be clever enough for the relation as well, but that may be a bit nicer long-form.

info.tier
 => "A" 
info.tier = "B"
 => "B" 
info.changed?
 => true 
info.tier = "A"
 => "A" 
info.changed?
 => false 
```

Reply:
My implementation prior to 8dc0a40 wasn't actually using the `.changed?` method – so the existence of `name_needs_updating?` is somewhat of a hang-over from that. Although now it's there, I do like the consistency of using the same pattern for both name and LDU.

It's very cool to see how smart the ActiveRecord dirty tracking is 🤓 I didn't know that!